compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -221,6 +221,97 @@ void addCPUBufferOpsTileAndVectorizePipeline(OpPassManager &passManager) {
   }
 }
 
+void addMatmulDoubleTilingExpertPassPipeline(OpPassManager &passManager) {
+  passManager.addPass(createVerifyLinalgTransformLegalityPass());
+
+  // Do first level of tiling and distribution.
+  passManager.addNestedPass<func::FuncOp>(createInsertDistributionInfoPass());
+  passManager.addNestedPass<func::FuncOp>(
+      createTileAndDistributeToWorkgroupsPass());
+  passManager.addPass(createCanonicalizerPass());
+  passManager.addPass(createCSEPass());
+  passManager.addNestedPass<func::FuncOp>(
+      createFoldAffineMinInDistributedLoopsPass());
```

Comment:
Would it make sense to refactor all these passes that we repeat for all the experts to a utility function?

Reply:
Yes, it does. I plan to do it after switching all pipelines to use upstream comprehensive bufferization, so the utility function can include converting to destination passing style pass. I'll do it in a separate PR.